main.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: false
+
+module Enumerable
+  def my_each(array)
+    num = []
+    x = 0
+    while x < array.length
+      num << yield(array[x])
+      x += 1
+    end
+    num
+  end
+
+  def my_each_with_index(array)
+    num = []
+    x = 0
+    while x < array.length
+      num << yield(array[x], x)
+      x += 1
+    end
+    num
+  end
+
+  def my_select(array)
+    num = []
+    x = 0
+    while x < array.length
+      if yield(array[x])
+        num << array[x]
+      end
+      x += 1
+    end
+    num
+  end
```

Comment:
```
def my_select
      return to_enum(method = :my_select) if !block_given?
      arr = []
       x = 0
     while x < self.length
       if yield(self[x])
         arr << self[x]
       end
       x += 1
      end
      arr
  end

```

Take a cue from this example, to understand what I meant in my earlier explanation on my_each, as regards to ``` self ```